fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2822,4 +2879,48 @@ public int doReindex(FHIROperationContext operationContext, OperationOutcome.Bui
         return FHIRValidator.validator().validate(resource);
     }
 
+    /**
+     * Validate an interaction for a specified resource type.
+     *
+     * @param interaction
+     *            the interaction to be performed
+     * @param resourceType
+     *            the resource type against which the interaction is to be performed
+     * @throws FHIROperationException
+     */
+    private void validateInteraction(String interaction, String resourceType) throws FHIROperationException {
+        List<String> interactions = null;
+
+        // Retrieve the interaction configuration
+        try {
+            StringBuilder defaultInteractionsConfigPath = new StringBuilder(FHIRConfiguration.PROPERTY_RESOURCES).append("/Resource/")
+                    .append(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_INTERACTIONS);
+            StringBuilder resourceSpecificInteractionsConfigPath = new StringBuilder(FHIRConfiguration.PROPERTY_RESOURCES).append("/")
+                    .append(resourceType).append("/").append(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_INTERACTIONS);
+
+            // Get the 'interactions' property
+            List<String> resourceSpecificInteractions = FHIRConfigHelper.getStringListProperty(resourceSpecificInteractionsConfigPath.toString());
+            if (resourceSpecificInteractions != null) {
+                interactions = resourceSpecificInteractions;
+            } else {
+                List<String> defaultInteractions = FHIRConfigHelper.getStringListProperty(defaultInteractionsConfigPath.toString());
+                if (defaultInteractions != null) {
+                    interactions = defaultInteractions;
+                }
+            }
+
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Allowed interactions: " + interactions);
+            }
+        } catch (Exception e) {
+            throw buildRestException("Error retrieving interactions configuration.", IssueType.UNKNOWN, IssueSeverity.ERROR);
+        }
+
+        // Perform validation of specified interaction against specified resourceType
+        if (interactions != null && !interactions.contains(interaction)) {
+            throw buildRestException("The requested interaction of type '" + interaction + "' is not allowed for resource type '" + resourceType + "'",
+                IssueType.BUSINESS_RULE, IssueSeverity.ERROR);
```

Comment:
Do you know what kind of error code this will result in?

Reply:
Looks like it will be an HTTP 400.
I wondering if that is suitable for all cases, or if we should attempt to make it a 404 if `fhirServer/resources/open` is set to false and the resource type is not listed.

Reply:
Wondering if 404 might be a little confusing, although they'd have the OperationOutcome indicating the error. Would 403 FORBIDDEN make sense, or is that more reserved for authorization type issues?

Reply:
Discussed with Lee - will use IssueType.NOT_FOUND (which maps to 404) if `fhirServer/resources/open` is set to false and resource type not listed,

Reply:
The updated logic looks OK to me